### PR TITLE
3/electrons

### PR DESCRIPTION
**Pull request by hagoptaminian**

### CODE REVIEW

File: src/components/common/CustomSwitch.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import Switch from '@material-ui/core/Switch';
+import Typography from '@material-ui/core/Typography';
+
+const useStyles = makeStyles((theme) => ({
+  wrapper: {
+    margin: theme.spacing(2, 0),
+    display: 'flex',
+    justifyContent: 'space-between',
+    width: '55%',
```
Comment (pyphilia) [inline src/components/common/CustomSwitch.js:13]: I don't really like this floating switch. Maybe we can display it really next to the label and later introduce a table to align every options / switches?

File: src/components/common/CustomSwitch.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  },
+  label: {
+    display: 'block',
+  },
+}));
+
+const CustomSwitch = ({ switchStatus, switchDispatch, switchLabel }) => {
```
Comment (pyphilia) [inline src/components/common/CustomSwitch.js:20]: What do you think about renaming it to `SwitchWithLabel`? `LabelledSwitch`?

Comment (pyphilia) [reply src/components/common/CustomSwitch.js:20]: Additionally I would rename `switchStatus` to `isChecked`, it sounds more logical for me.

Comment (pyphilia) [reply src/components/common/CustomSwitch.js:20]: And `switchDispatch` to `onToggle`

File: src/utils/utils.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+  // since last ion will not have padding to its right
+  const totalWidthOfPositiveIons =
+    numberOfPositiveIons * widthOfPositiveIonWithPadding -
+    distanceBetweenPositiveIons;
+
+  const excessWidth = stageWidth - totalWidthOfPositiveIons;
+  const leftIndent = excessWidth / 2;
+
+  return [...Array(numberOfPositiveIons)].map(
```
Comment (pyphilia) [inline src/utils/utils.js:26]: Do you still need to destruct the array?

**Review by pyphilia — CHANGES_REQUESTED** [review]
Very nice! I really like these little electrons 👍

File: src/utils/utils.js (hunk 3 above)
Comment (hagoptaminian) [reply src/utils/utils.js:26]: yes, turns out this is necessary. `Array(numberOfPositiveIons).map(...)` doesn't work

**Pull request merged**